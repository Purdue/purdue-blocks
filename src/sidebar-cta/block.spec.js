// title-nav test
jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    blockStartup,
    clickCheckbox,
    clickRadio,
    clickElementByText,
} from '../test-helpers'

const block = {blockTitle: 'Sidebar CTA', blockName: 'purdue-blocks/sidebar-cta'}

describe( 'ðŸ”¬ Sidebar CTA Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'Block should be available.', async () => {
        await insertBlock( block.blockTitle )


        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        test('ðŸ”Ž Header Text field updates output correctly', async () => {
            await blockStartup(block)

            const typeString = "Header text test."

            await page.focus( '.block-editor-rich-text__editable.header' );
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( editedContent.includes(`<p class="header">${typeString}</p>`)).toBe(true)
            expect( editedContent).toMatchSnapshot()
        })
        test('ðŸ”Ž Content Text field updates output correctly', async () => {
            await blockStartup(block)

            const typeString = "Content text test."

            await page.focus( '.block-editor-rich-text__editable.content' );
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( editedContent.includes(`<p class="content">${typeString}</p>`)).toBe(true)
            expect( editedContent).toMatchSnapshot()
        })
        test('ðŸ”Ž Button Text field updates output correctly', async () => {
            await blockStartup(block)

            const typeString = "Button Text"
            
            await page.focus( '.block-editor-rich-text__editable.cta-button' );
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( editedContent.includes(`<a href="" class="cta-button" target="_self" rel="noopener noreferrer">${typeString}</a>`)).toBe(true)
            expect( editedContent).toMatchSnapshot()
        })    
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        describe('ðŸ”¬ Background color controls', () => {
            test('ðŸ”Ž Black background option is slectable', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'black')
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"backgroundColor":"black"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
            test('ðŸ”Ž Gray background option is slectable', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'gray')
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"backgroundColor":"gray"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
            test('ðŸ”Ž White background option is slectable', async () => {
                await blockStartup(block)

                await clickRadio('Background Color', 'white')
                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"backgroundColor":"gray"`)).toBe(false)
                expect( editedContent.includes(`"backgroundColor":"black"`)).toBe(false)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })

        describe('ðŸ”¬ Button Link Settings', () => {
            test('ðŸ”Ž Link URL field updates output correctly ', async () => {
                await blockStartup(block)

                const typeString = "https://www.purdue.edu"
                await clickElementByText('label', 'Link URL')
                await page.keyboard.type(typeString, {delay: 10})

                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"ctaUrl":"${typeString}"`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })

            test('ðŸ”Ž Link can be opened in new tab', async () => {
                await blockStartup(block)

                await clickCheckbox('Open link in new tab?')

                const editedContent = await getEditedPostContent()
                expect( editedContent.includes(`"external":true`)).toBe(true)
                expect( await getEditedPostContent() ).toMatchSnapshot()
            })
        })
    })
})