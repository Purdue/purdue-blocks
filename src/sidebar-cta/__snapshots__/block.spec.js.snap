// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Sidebar CTA Block Block should be available. 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Block Editor Fields 🔎 Button Text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\">Button Text</a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Block Editor Fields 🔎 Content Text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><p class=\\"content\\">Content text test.</p><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Block Editor Fields 🔎 Header Text field updates output correctly 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><p class=\\"header\\">Header text test.</p><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 Black background option is slectable 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta {\\"backgroundColor\\":\\"black\\"} -->
<div class=\\"purdue-blocks-sidebar-cta purdue-blocks-sidebar-cta-black
      \\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 Gray background option is slectable 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta {\\"backgroundColor\\":\\"gray\\"} -->
<div class=\\"purdue-blocks-sidebar-cta
       purdue-blocks-sidebar-cta-gray\\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Side Panel Settings 🔬 Background color controls 🔎 White background option is slectable 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Side Panel Settings 🔬 Button Link Settings 🔎 Link URL field updates output correctly  1`] = `
"<!-- wp:purdue-blocks/sidebar-cta {\\"ctaUrl\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><a href=\\"https://www.purdue.edu\\" class=\\"cta-button\\" target=\\"_self\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;

exports[`🔬 Sidebar CTA Block 🔬 Side Panel Settings 🔬 Button Link Settings 🔎 Link can be opened in new tab 1`] = `
"<!-- wp:purdue-blocks/sidebar-cta {\\"external\\":true} -->
<div class=\\"purdue-blocks-sidebar-cta
      \\"><a href=\\"\\" class=\\"cta-button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"></a></div>
<!-- /wp:purdue-blocks/sidebar-cta -->"
`;
